client/src/hooks/useLogin.ts

```diff
@@ -0,0 +1,39 @@
+
+import axios from 'axios';
+import { useState } from 'react';
+
+export const useLogin = () => {
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const login = async (email: string, password: string) => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      const response = await axios.post('http://localhost:3000/v1/login', {
+        email,
+        password
+      });
+
+      localStorage.setItem('token', response.data.token);
+      setLoading(false);
+      
+      return response.data.user;
```

Comment:
Maybe we can use some schema validation or some mechanism of validation here, to make sure the response is what we want. Check [zod](https://zod.dev/)